### PR TITLE
comid: Add OVMF metadata to the Measurement type

### PR DESCRIPTION
**Pull request by jraman567**
SEV-SNP launch digest is a function of OVMF metadata, among others. Therefore, this patch adds support to add it as a reference value to Veraison.

We need this reference value to compute the SEV-SNP launch digest on-demand.

### CODE REVIEW

**Comment by jraman567** [comment]
Hi there,

Please review this pull request and let me know if I need to make any changes.

I ran make lint; make presubmit.

lint didn't report any errors. However, it needs Golang version 1.21 to complete; golangci-lint needs a minimum of 1.21.

presubmit reported a few errors. However, they are existing errors.

Thank you, @thomas-fossati for the CI patch.

Thank you!

**Comment by jraman567** [comment]
Hi @yogeshbdeshpande @thomas-fossati @setrofim ,

I just sent a pull request to add OVMF metadata to CoRIM reference values. Please let me know if it looks good.

Thank you, @thomas-fossati , for fixing the CI issue.

--
Jag

File: comid/measurement.go [hunk 1]
```diff
@@ -343,18 +344,19 @@ func RegisterMkeyType(tag uint64, factory IMkeyFactory) error {
...
+	Flags              *FlagsMap             `cbor:"3,keyasint,omitempty" json:"flags,omitempty"`
+	RawValue           *RawValue             `cbor:"4,keyasint,omitempty" json:"raw-value,omitempty"`
+	RawValueMask       *[]byte               `cbor:"5,keyasint,omitempty" json:"raw-value-mask,omitempty"`
+	MACAddr            *MACaddr              `cbor:"6,keyasint,omitempty" json:"mac-addr,omitempty"`
+	IPAddr             *net.IP               `cbor:"7,keyasint,omitempty" json:"ip-addr,omitempty"`
+	SerialNumber       *string               `cbor:"8,keyasint,omitempty" json:"serial-number,omitempty"`
+	UEID               *eat.UEID             `cbor:"9,keyasint,omitempty" json:"ueid,omitempty"`
+	UUID               *UUID                 `cbor:"10,keyasint,omitempty" json:"uuid,omitempty"`
+	IntegrityRegisters *IntegrityRegisters   `cbor:"14,keyasint,omitempty" json:"integrity-registers,omitempty"`
+	OvmfMetadata       *ovmf.MetadataWrapper `cbor:"15,keyasint,omitempty" json:"ovmf-metadata,omitempty"`
```
Comment (yogeshbdeshpande) [inline comid/measurement.go:359]: This needs to be done as extension work, not natively inside CoRIM.

**Review by thomas-fossati — COMMENTED** [review]
Hi @jraman567, thanks very much for your contribution.

From the CoRIM perspective, OvmfMetadata is a profile-specific extension.  Since it's not part of the base spec, it should be handled using `veraison/corim`'s [extensions framework](https://github.com/veraison/corim?tab=readme-ov-file#extending-corimcomid) rather than adding it directly into `struct Mval`.

Of course, we can work with you to make the needed changes.

**Pull request closed without merging**